From 6ec1fbd4b22eaa38376228a9052fcd61412ebae6 Mon Sep 17 00:00:00 2001
From: Daniel Leung <daniel.leung@intel.com>
Date: Fri, 13 Dec 2013 09:10:40 -0800
Subject: kernel: fix build issue when no EXTRA_KERNEL_MODULES defined

When there is no EXTRA_KERNEL_MODULES defined, the make is invoked again
with no parameters. This causes building of everything again and
the build will fail.

This adds a conditional statement to skip calling make if there is
no EXTRA_KERNEL_MODULES defined.

(Not upstream candidate since the kernel build mechanism is not
 in AOSP.)

Category: AOSP Enhancement
Domain: AOSP-build
Upstream-Candidate: no
Change-Id: I2498957c34fdcf228b8bcac6e49263127fdb51b4
Signed-off-by: Daniel Leung <daniel.leung@intel.com>
---
 core/kernel.mk |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/core/kernel.mk b/core/kernel.mk
index 77a7f80..8aaa638 100644
--- a/core/kernel.mk
+++ b/core/kernel.mk
@@ -164,7 +164,7 @@ endef
 define make-modules
 	$(mk_kernel) INSTALL_MOD_PATH=$(1) modules_install
 	$(foreach item,$(dir $(EXTERNAL_KERNEL_MODULES_TO_INSTALL)),$(call make-ext-module,$(item),$(1)))
-	+ $(hide) $(MAKE) $(EXTRA_KERNEL_MODULES)
+	+ $(if $(EXTRA_KERNEL_MODULES), $(hide) $(MAKE) $(EXTRA_KERNEL_MODULES))
 	$(hide) rm -f $(1)/lib/modules/*/{build,source}
 	$(hide) cd $(1)/lib/modules && find -type f -print0 | xargs -t -0 -I{} mv {} .
 endef
-- 
1.7.4.1

